cdc/generation.cc

```diff
@@ -310,7 +310,14 @@ db_clock::time_point make_new_cdc_generation(
     auto ts = db_clock::now() + (
             (for_testing || ring_delay == milliseconds(0)) ? milliseconds(0) : (
                 2 * ring_delay + duration_cast<milliseconds>(generation_leeway)));
-    sys_dist_ks.insert_cdc_topology_description(ts, std::move(gen), { tm.count_normal_token_owners() }).get();
+    try {
+        sys_dist_ks.insert_cdc_topology_description(ts, std::move(gen), { tm.count_normal_token_owners() }).get();
+    } catch (...) {
```

Comment:
We only want to make this optional in the replace-node or rolling-upgrade case, but not in the node-bootstrap case. For node bootstrap changing the generation is mandatory.

Reply:
Hence I think this `try`+`catch` should be outside the function, in the `if (!_cdc_streams_ts) {` path (line 552 in storage service).

Reply:
done